### PR TITLE
[MDS-6129] project summary status: frontend

### PR DESCRIPTION
**Pull request by taraepp**
## Objective 
- fixing issues
- adding automatic tests, because manual tests are too time-consuming

[MDS-6129](https://bcmines.atlassian.net/browse/MDS-6129)

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

File: services/common/src/components/projectSummary/ProjectDates.spec.tsx [hunk 1]
```diff
@@ -28,9 +28,9 @@ describe("ProjectDates", () => {
         <FormWrapper
           name={FORM.ADD_EDIT_PROJECT_SUMMARY}
           initialValues={MOCK.PERMITS}
```
Comment (taraepp) [inline services/common/src/components/projectSummary/ProjectDates.spec.tsx]: Some of these, the form initialValues are very wacky.

File: services/common/src/components/forms/RenderFileUpload.tsx [hunk 2]
```diff
@@ -197,9 +196,8 @@ export const FileUpload: FC<FileUploadProps> = ({
         )
       ) {
         notification.error({
-          message: `Failed to upload ${file && file.name ? file.name : ""}: ${
-            errorMessage ? errorMessage : err
-          }`,
+          message: `Failed to upload ${file?.name ?? ""}: ${errorMessage ?? err
```
Comment (matbusby-fw) [inline services/common/src/components/forms/RenderFileUpload.tsx]: Super minor, but if the file.name is null would this not render as "Failed to upload : {errorMessage}". Noting the presence and placement of the colon.

File: services/common/src/components/projects/projectUtils.ts [hunk 3]
```diff
@@ -1,44 +1,64 @@
...
+import { memoize } from "lodash";
 
-export const isFieldDisabled = (
-    systemFlag,
-    projectSummaryStatusCode = "",
-    isAuthorizationPage = false
-) => {
-    // Return false if status = "" => "Not Started"
-    if (!projectSummaryStatusCode) return false;
+export const areFieldsDisabled = memoize((systemFlag: SystemFlagEnum, projectSummaryStatusCode: string) => {
```
Comment (matbusby-fw) [inline services/common/src/components/projects/projectUtils.ts:4]: Nice solution!

**Review by matbusby-fw — DISMISSED** [review]
Nice!  This looks like it should make things MUCH more consistent.

File: services/common/src/components/projectSummary/ProjectSummaryFormComponents.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+
+const amsAuthTypes = ['AIR_EMISSIONS_DISCHARGE_PERMIT', 'EFFLUENT_DISCHARGE_PERMIT', 'REFUSE_DISCHARGE_PERMIT'];
+const project = { project_lead_party_guid: "project_lead_party_guid" };
+const formattedProjectSummary = formatProjectSummary(MOCK.PROJECT_SUMMARY, project, amsAuthTypes);
+
+const mine_guid = formattedProjectSummary.mine_guid;
+// this prevents console spam of "not wrapped in act()"
+// by making the permits belong to the mine and therefore have it be loaded
+const permits = MOCK.PERMITS.map((permit) => ({ ...permit, mine_guid }));
```
Comment (matbusby-fw) [inline services/common/src/components/projectSummary/ProjectSummaryFormComponents.spec.tsx:34]: Oh nice.  That's an annoying message!

**Review by matbusby-fw — APPROVED** [review]
Nice enhancement of the tests!

**Review by asinn134 — APPROVED** [review]

**Review by simensma-fresh — APPROVED** [review]

**Pull request merged**